zerver/tests/test_outgoing_webhook_system.py

```diff
@@ -0,0 +1,108 @@
+    def test_retry_request(self, mock_request_retry):
+        # type: (mock.Mock) -> None
+        response = ResponseMock(500, {"message": "testing"})
+        with mock.patch('requests.request', return_value=response):
+            do_rest_call(rest_operation, None, None)
+            self.assertTrue(mock_request_retry.called)
+
+    @mock.patch('zerver.lib.outgoing_webhook.fail_with_message')
+    def test_fail_request(self, mock_fail_with_message):
+        # type: (mock.Mock) -> None
+        response = ResponseMock(400, {"message": "testing"})
+        with mock.patch('requests.request', return_value=response):
+            do_rest_call(rest_operation, None, None)
+            self.assertTrue(mock_fail_with_message.called)
+
+    @mock.patch('logging.info')
+    @mock.patch('requests.request', side_effect=timeout_error)
+    @mock.patch('zerver.lib.outgoing_webhook.request_retry')
+    def test_timeout_request(self, mock_request_retry, mock_requests_request, mock_logger):
+        # type: (mock.Mock, mock.Mock, mock.Mock) -> None
+        do_rest_call(rest_operation, {"command": "", "service_name": ""}, None)
+        self.assertTrue(mock_request_retry.called)
+
+    @mock.patch('logging.exception')
+    @mock.patch('requests.request', side_effect=request_exception_error)
+    @mock.patch('zerver.lib.outgoing_webhook.fail_with_message')
+    def test_request_exception(self, mock_fail_with_message, mock_requests_request, mock_logger):
+        # type: (mock.Mock, mock.Mock, mock.Mock) -> None
+        do_rest_call(rest_operation, {"command": ""}, None)
+        self.assertTrue(mock_fail_with_message.called)
+
+
+class TestEventPublishedToQueue(ZulipTestCase):
+
+    @mock.patch('zerver.lib.actions.queue_json_publish')
+    def test_event_flow(self, mock_queue_json_publish):
+        # type: (mock.Mock) -> None
+        user_profile = get_user_profile_by_email("othello@zulip.com")
+        bot_profile = do_create_user(email="foo-bot@zulip.com",
+                                     password = "test",
+                                     realm = get_realm_by_email_domain("zulip.com"),
+                                     full_name = "FooBot",
+                                     short_name = "foo-bot",
+                                     bot_type = UserProfile.OUTGOING_WEBHOOK_BOT,
+                                     bot_owner = user_profile)
+        content = u'@**FooBot** foo bar!!!'
+        # TODO: In future versios this won't be required
+        self.subscribe_to_stream(bot_profile.email, "Denmark")
+
+        def check_values_passed(queue_name, trigger_event, x):
+            # type: (Any, Union[Mapping[Any, Any], Any], Callable[[Any], None]) -> None
+            self.assertEqual(queue_name, "outgoing_webhooks")
+            self.assertEqual(trigger_event['user_profile_id'], bot_profile.id)
+            self.assertEqual(trigger_event["message"]["sender_email"], user_profile.email)
```

Comment:
Thanks for adding this test!  Would it be helpful to also check the value of `['trigger']` here?